A conformance test is failed if master node doesn't have taints

**What happened**:

The conformance test "validates resource limits of pods that are allowed to run" is failed if master node doesn't have taints.

**What you expected to happen**:

All conformance tests should pass even if a master node doesn't have taints.

**How to reproduce it (as minimally and precisely as possible)**:

If a master node has taints like many cases, the test can pass like the following:
```
$ kubectl describe node k8s-master | grep Taint
Taints:             key=value:NoSchedule
$ go run hack/e2e.go -- --provider=skeleton --test --test_args="--ginkgo.focus=validates\sresource\slimits\sof\spods\sthat\sare\sallowed"
...
Ran 1 of 4798 Specs in 13.629 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 4797 Skipped
PASS
```
However if the master node doesn't have any taints, the test cannot pass like:
```
$ kubectl describe node k8s-master | grep Taint
Taints:             <none>
$ go run hack/e2e.go -- --provider=skeleton --test --test_args="--ginkgo.focus=validates\sresource\slimits\sof\spods\sthat\sare\sallowed"
...
Summarizing 1 Failure:

[Fail] [sig-scheduling] SchedulerPredicates [Serial] [It] validates resource limits of pods that are allowed to run  [Conformance]
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduling/predicates.go:738

Ran 1 of 4798 Specs in 142.835 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 4797 Skipped
--- FAIL: TestE2E (142.85s)
```

**Anything else we need to know?**:

I am guessing this issue can be related to https://github.com/kubernetes/kubernetes/issues/80177

**Environment**:
- Kubernetes version (use `kubectl version`): v1.15.1
- Cloud provider or hardware configuration: OpenStack
- OS (e.g: `cat /etc/os-release`): Ubuntu 18.04.2 LTS
- Kernel (e.g. `uname -a`): 4.15.0-46-generic
- Install tools: kubeadm
- Network plugin and version (if this is a network-related bug):
- Others:

--- a/test/e2e/scheduling/predicates.go
+++ b/test/e2e/scheduling/predicates.go
@@ -87,10 +87,17 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
 		var err error
 
 		framework.AllNodesReady(cs, time.Minute)
-		masterNodes, nodeList, err = e2enode.GetMasterAndWorkerNodes(cs)
+
+		// NOTE: Here doesn't get nodeList for supporting a master nodes which can host workload pods.
+		masterNodes, _, err = e2enode.GetMasterAndWorkerNodes(cs)
 		if err != nil {
 			e2elog.Logf("Unexpected error occurred: %v", err)
 		}
+		nodeList, err = e2enode.GetReadySchedulableNodesOrDie(cs)
+		if err != nil {
+			e2elog.Logf("Unexpected error occurred: %v", err)
+		}
+
 		// TODO: write a wrapper for ExpectNoErrorWithOffset()
 		framework.ExpectNoErrorWithOffset(0, err)
 

